torch/ao/quantization/pt2e/numeric_debugger.py

```diff
@@ -11,3 +20,188 @@ def generate_numeric_debug_handle(graph_module: GraphModule) -> None:
         if node.op != "placeholder" and NUMERIC_DEBUG_HANDLE_KEY not in node.meta:
             node.meta[NUMERIC_DEBUG_HANDLE_KEY] = unique_id
             unique_id += 1
+
+
+class OutputLogger(torch.nn.Module):
+    """
+    Base class for capturing output values.
+    """
+
+    # Mark as impure so that calls to it will not be removed during DCE.
+    _is_impure = True
+
+    def __init__(
+        self,
+        node_name: str,
+        nn_module_stack: object,
```

Comment:
You could make this default to None, in case other users don't have a stack available. It's an optional part of the logger, just to make identifying stuff easier. Same with `node_name` too, only `debug_handle` is required